project1/src/main/java/uk/co/cpsd/javaproject1/Animal.java

```diff
@@ -7,11 +7,13 @@ public abstract class Animal {
     protected int x;
     protected int y;
     protected int energyLevel;
+    protected int id;
 
-    public Animal(int x,int y, int energyLevel){
+    public Animal(int x,int y, int energyLevel,int animalId){
         this.x=x;
         this.y=y;
         this.energyLevel=energyLevel;
+        this.id=animalId;
```

Comment:
This might be a good opportunity to use static methods in classes, and possibly even AtomicInteger.  Make sure each Animal has a unique Id by setting it up in the constructor